pkg/build/controller/build/build_controller.go

```diff
@@ -2023,25 +2067,41 @@ func (bc *BuildController) getAdditionalTrustedCAData(config *configv1.Image) (m
 	return additionalCA.Data, nil
 }
 
-func (bc *BuildController) createBuildRegistriesConfigData(config *configv1.Image) (string, error) {
+func (bc *BuildController) createBuildRegistriesConfigData(config *configv1.Image, policy *operatorv1alpha1.ImageContentSourcePolicy) (string, error) {
 	registriesConfig := config.Spec.RegistrySources
-	if len(registriesConfig.InsecureRegistries) == 0 {
+	if len(registriesConfig.InsecureRegistries) == 0 && (policy == nil || len(policy.Spec.RepositoryDigestMirrors) == 0) {
```

Comment:
I can’t see how not building the configuration is correct, if the code below sets `docker.io` in the search path.

Reply:
Yeah ... following the previous logic.  I'll have to defer to @bparees or @adambkaplan on the history / rationale here.